File: app/src/main/java/com/shizuku/runner/plus/ui/activity/BaseActivity.java
package com.shizuku.runner.plus.ui.activity;

import android.graphics.Color;
import android.os.Bundle;
import android.view.Window;

import rikka.core.util.ResourceUtils;
import rikka.material.app.MaterialActivity;

public class BaseActivity extends MaterialActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {

        //设置主题
        if (ResourceUtils.isNightMode(getResources().getConfiguration()))
            setTheme(rikka.material.R.style.Theme_Material3_DynamicColors_Dark_Rikka);
        else
            setTheme(rikka.material.R.style.Theme_Material3_DynamicColors_Light_Rikka);
        super.onCreate(savedInstanceState);

        //隐藏actionBar栏
        if (getSupportActionBar() != null)
            getSupportActionBar().hide();
    }


    @Override
    public void onApplyTranslucentSystemBars() {
        super.onApplyTranslucentSystemBars();

        //设置状态栏导航栏透明
        Window window = getWindow();
        window.setStatusBarColor(Color.TRANSPARENT);
        window.setNavigationBarColor(Color.TRANSPARENT);
    }
}

File: app/src/main/java/yangfentuozi/runner/app/base/BaseActivity.kt
package yangfentuozi.runner.app.base

import android.graphics.Color.TRANSPARENT
import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.SystemBarStyle
import androidx.activity.enableEdgeToEdge
import yangfentuozi.runner.app.App


open class BaseActivity() : ComponentActivity(),
    BaseDialogBuilder.IsDialogShowing {
    override var isDialogShowing: Boolean = false

    lateinit var mApp: App

    override fun onCreate(savedInstanceState: Bundle?) {
        mApp = application as App
        mApp.addActivity(this)
        super.onCreate(savedInstanceState)
        // 设置状态栏导航栏透明
        enableEdgeToEdge(
            statusBarStyle = SystemBarStyle.auto(TRANSPARENT, TRANSPARENT),
            navigationBarStyle = SystemBarStyle.auto(TRANSPARENT, TRANSPARENT)
        )
    }

    override fun onDestroy() {
        mApp.removeActivity(this)
        super.onDestroy()
    }
}
